### PR TITLE
NMS-12938: TSS: Cortex Plugin: Use REST API for reading timeseries

### PR DESCRIPTION
**Pull request by patrick-schweizer**
Jira: https://issues.opennms.org/browse/NMS-12938

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -14,15 +14,16 @@
...
+        <junit.version>4.13.1</junit.version>
         <karaf.version>4.1.5</karaf.version>
         <log4j.version>2.13.3</log4j.version>
         <metrics.version>4.1.2</metrics.version>
         <mockito.version>2.18.0</mockito.version>
         <okio.bundle.version>1.14.0_1</okio.bundle.version>
         <okhttp.version>3.10.0</okhttp.version>
         <okhttp.bundle.version>3.10.0_2</okhttp.bundle.version>
-        <opennms.api.version>0.5.1-SNAPSHOT</opennms.api.version>
+        <opennms.api.version>0.5.2-SNAPSHOT</opennms.api.version>
```
Comment (patrick-schweizer) [inline pom.xml:26]: @j-white I had to bump the version in order to get the Integration tests running. I am not sure of the implications:
The requirement only affects the tests but it asks now for a higher version not necessarily provided by the Opennms installation?
What do you think we should do?

**Review by indigo423 — COMMENTED** [review]
Happy you found time to address this issue :) The data is now shown but I've noticed if you pick a larger time window, the data disappear. I've maded a dashboard with our HELM Performance data source and the Prometheus Data Source with the same metric as a comparison. As you can see the last 2 days and last 7 days time window works perfectly. But when I pick the last 30 days there is no data and the Prometheus data source shows data.

2-days
7-days
30-days

**Comment by indigo423** [comment]
What's also interesting, even if they read exactly the same data, the graphs are slightly different:

**Comment by mfuhrmann** [comment]
Nice!

**Comment by patrick-schweizer** [comment]
@indigo423 thank you for testing :-)
I take it we are not where we should be. I did a bit more testing and found some oddities:

First a view words on what the TSS does: we normally want to get the raw data back from the data base in order to do our own aggregation. The reason for this is that we want to aggregate in the exact same way as Newts so that the display between using Newts and the TSS is the same. But that doesn't necessarily mean it gives the exact same graph as Grafana. 

On top of that it turns out that the Cortex Querier doesn't give us the possibility to query raw data. The range query has a mandatory field "step": https://prometheus.io/docs/prometheus/latest/querying/api/#range-queries which defines the resolution.

The maximum number of data points a range query can deliver is 11,000. I tried to adopt the step so that we should get always 11,000 data points with a resolution of a minimum of 1 second (smallest resolution by Cortex). But this doesn't seem to work as expected, see this test: 
https://github.com/OpenNMS/opennms-cortex-tss-plugin/pull/7/files#diff-066cf16de53e42426251f8e9c6c01173b4656c00b280c9fd89a15a929bf21705R89

We get way less samples and they are oddly spaced time wise.

So what are the next steps?
The TSS provides the possibility to move the aggregation from the TSS to the data base. That should give us the same results as Grafana uses. I would attempt to do that next. Then we can compare the query and its results from Grafana to Cortex and the query from the plugin to Cortex.

**Pull request merged**